src/fu-device-list.c

```diff
@@ -713,6 +713,16 @@ fu_device_list_replace(FuDeviceList *self, FuDeviceItem *item, FuDevice *device)
 		fu_device_set_parent(device, parent);
 	}
 
+	/* copy the children */
+	if (fu_device_get_children(item->device)->len > 0) {
+		GPtrArray *children = fu_device_get_children(item->device);
```

Comment:
major nit; but I think that since you're always going to be evaluating children at least once, just make the `GPtrArray *children = fu_device_get_children(item->device);` at top level scope for function.  Then you won't need to evaluate it twice.

Reply:
Certainly, made the change. :)